src/site/filters/liquid.js

```diff
@@ -610,6 +610,50 @@ module.exports = function registerFilters() {
     return _.replace(url, 'youtu.be', 'youtube.com/embed');
   };
 
+  liquid.filters.deriveCLPTotalSections = (
+    maxSections,
+    fieldClpVideoPanel,
+    fieldClpSpotlightPanel,
+    fieldClpStoriesPanel,
+    fieldClpResourcesPanel,
+    fieldClpEventsPanel,
+    fieldClpFaqPanel,
+    fieldBenefitCategories,
+  ) => {
+    // Start total sections as max number.
+    let clpTotalSections = maxSections;
+
+    if (!fieldClpVideoPanel) {
+      clpTotalSections -= 1;
+    }
+
+    if (!fieldClpSpotlightPanel) {
+      clpTotalSections -= 1;
+    }
+
+    if (!fieldClpStoriesPanel) {
+      clpTotalSections -= 1;
+    }
+
+    if (!fieldClpResourcesPanel) {
+      clpTotalSections -= 1;
+    }
+
+    if (!fieldClpEventsPanel) {
+      clpTotalSections -= 1;
+    }
+
+    if (!fieldClpFaqPanel) {
+      clpTotalSections -= 1;
+    }
```

Comment:
nothing wrong with the way you're doing it already but potentially we could stick all these in an array and filter out the null values ->

```
[
 fieldClpVideoPanel,
 fieldClpSpotlightPanel, 
// ...
].filter(panel => !!panel).length
```

Reply:
Done! 🙂